init/eessi_environment_variables

```diff
@@ -38,6 +38,45 @@ if [ -d $EESSI_PREFIX ]; then
           break
         fi
       done
+
+      # we need to make sure that errexit shell option (set -e) is not enabled,
+      # since archdetect will produce non-zero exit code if no accelerator was found
+      if [[ "$-" =~ e ]]; then
+        errexit_shell_option_set='yes'
+        set +e
+      else
+        errexit_shell_option_set='no'
+      fi
+
+      # to be able to grab exit code of archdetect trying to detect accelerators,
+      # we can not run it via $(...), so we have to redirect the output to a temporary file
+      tmpout=$(mktemp)
+      ${EESSI_INIT_DIR_PATH}/eessi_archdetect.sh accelpath 2>&1 > $tmpout
+      accelpath_exit_code=$?
+
+      if [[ "$errexit_shell_option_set" == "yes" ]]; then
+        set -e
+      fi
```

Comment:
You're going to hate me for this but if we call `archdetect` using `source_sh` in Lmod, I think we will still need a mechanism to be able to switch off returning an error. I'll look into it as part of the PR for `init/modules/EESSI/2023.06`

Reply:
Heh, ok...

Yeah, let's tackle that in a subsequent PR where we make the EESSI module accelerator-aware